### PR TITLE
Fix(eos_validate_state): ANTA Add a check to AvdTestLLDPTopology for shutdown interfaces

### LINKED ISSUE #3188: Fix(eos_validate_state): In my tests, when I set `dc2-leaf1c` and `dc2-leaf2c` (the L2 leave) and set `shutdown_interfaces_towards_undeployed_peers: True`
In my tests, when I set `dc2-leaf1c` and `dc2-leaf2c` (the L2 leave) and set `shutdown_interfaces_towards_undeployed_peers: True`, I noticed:
- the interface check on interfaces towards the L2 Leafs were set to check for 'down' status **as expected**
- The LLDP neighborchip check on L3 Leafs still expected to find the L2 Leaf downstream **I think this is a bug**

_Originally posted by @chetryan in https://github.com/aristanetworks/ansible-avd/issues/3171#issuecomment-1740914826_

### PR DESCRIPTION
**Pull request by carl-baillargeon**
## Change Summary

- `AvdTestLLDPTopology` will now skip the `VerifyLLDPNeighbors` ANTA test for an ethernet interface that is shutdown in the structured_config. This covers the use case where `shutdown_interfaces_towards_undeployed_peers` is set to True.

- Added unit tests for `AvdTestLLDPTopology`
- Adjusted molecule scenario

## Related Issue(s)

Fixes #3188

## Component(s) name

`arista.avd.eos_validate_state` in **ANTA mode**

## Proposed changes
Added the following check when looping over the `ethernet_interfaces`:
```
if ethernet_interface["shutdown"]:
    LOGGER.info("Ethernet interface '%s' is shutdown. 'VerifyLLDPNeighbors' is skipped for this interface.", ethernet_interface["name"])
    continue
```

## How to test
Tested with `eos_validate_state` molecule. `VerifyLLDPNeighbors` test is now removed from leafs shutdown interfaces in the test catalog.

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@
 # Default gateway used for the management interface
 mgmt_gateway: 172.16.1.1
 
+### DNS Domain ###
+custom_structured_configuration_dns_domain: dc1.local
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml:8]: Is this required for this test? If so we should guard that somehow.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py [hunk 2]
```diff
@@ -202,36 +202,44 @@ def test_definition(self) -> dict | None:
...
         required_vars = ["name", "shutdown", "peer", "peer_interface"]
 
-        for ethernet_interface in ethernet_interfaces:
+        for idx, ethernet_interface in enumerate(ethernet_interfaces, start=1):
             try:
                 for var in required_vars:
                     get(ethernet_interface, var, required=True)
+            except AristaAvdMissingVariableError as e:
+                LOGGER.warning("Ethernet interface entry #%d from the 'ethernet_interfaces' data model is missing the variable '%s'.", idx, str(e))
+                continue
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py:211]: You know the primary key is "name" here, so presenting the value of "name" would be much more helpful to the user instead of the index.

File: ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml (hunk 1 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml:8]: Not required. If it's present it means the device will use his FQDN in LLDP so we need to give the proper input. I added the variable only in DC1.yml and not DC2.yml in molecule to test both scenarios.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml:8]: perfect

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py (hunk 2 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py:211]: It covers if `name` is missing from the structured_config.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py:211]: It would be better to just implement schema validation.

Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py:211]: I agree, that will clean-up a lot of code. Issue #3259 opened. I will work on that later but for now let's guard everything in code.

File: ansible_collections/arista/avd/tests/unit/roles/eos_validate_state/tests/test_avdtestrouting.py [hunk 3]
```diff
@@ -222,5 +222,25 @@
         "expected_log": "Peer group 'IPv4-UNDERLAY-PEERS' from the 'peer_groups' list of the 'router_bgp' data model is missing the variable 'type'.",
         "expected_log_level": "WARNING",
     },
+    {
+        "test_name": "missing-the-only-neighbor-ip-address",
+        "test_module": "AvdTestBGP",
```
Comment (gmuloc) [inline ansible_collections/arista/avd/tests/unit/roles/eos_validate_state/tests/test_avdtestrouting.py:227]: Looks like it should be in another PR ? I am fine to leave it in there if it is an additional test for something already implemented though.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 4]
```diff
@@ -163,4 +163,4 @@ def add_verify_peers_test(description: str, afi: str, bgp_neighbor_ip: str, safi
             elif neighbor_peer_group_type == "evpn":
                 add_verify_peers_test(description="bgp evpn peer state established (evpn)", afi="evpn", bgp_neighbor_ip=bgp_neighbor_ip)
 
-        return {self.anta_module: anta_tests}
+        return {self.anta_module: anta_tests} if anta_tests.get("bgp") else None
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py:166]: BGP stuff?

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1b-catalog.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 anta.tests.connectivity:
 - VerifyLLDPNeighbors:
     neighbors:
-    - neighbor_device: dc1-leaf1a
+    - neighbor_device: dc1-leaf1a.dc1.local
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1b-catalog.yml:4]: so given that this leaf is not deployed :) should we REALLY be running any tests ? ;) - this should probably be a new PR to tackle this very simple use case

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1b-catalog.yml:4]: doh - https://github.com/aristanetworks/ansible-avd/pull/3272

File: ansible_collections/arista/avd/tests/unit/roles/eos_validate_state/tests/test_avdtestrouting.py (hunk 3 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/tests/unit/roles/eos_validate_state/tests/test_avdtestrouting.py:227]: Right this is an additional test.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py (hunk 4 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py:166]: This is a fix on something already implemented.

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1b-catalog.yml (hunk 5 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc1-leaf1b-catalog.yml:4]: :)

**Review by gmuloc — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
